runtime/moonbase/src/governance/tracks.rs

```diff
@@ -0,0 +1,286 @@
+			min_enactment_period: 28 * DAYS,
+			min_approval: APP_BIG_TIPPER,
+			min_support: SUP_BIG_TIPPER,
+		},
+	),
+	(
+		16,
+		pallet_referenda::TrackInfo {
+			name: "small_spender",
+			max_deciding: 50,
+			decision_deposit: 50 * KILOUNIT * SUPPLY_FACTOR,
+			prepare_period: 4,
+			decision_period: 28 * DAYS,
+			confirm_period: 12 * HOURS,
+			min_enactment_period: 28 * DAYS,
+			min_approval: APP_SMALL_SPENDER,
+			min_support: SUP_SMALL_SPENDER,
+		},
+	),
+	(
+		17,
+		pallet_referenda::TrackInfo {
+			name: "medium_spender",
+			max_deciding: 20,
+			decision_deposit: 150 * KILOUNIT * SUPPLY_FACTOR,
+			prepare_period: 4,
+			decision_period: 28 * DAYS,
+			confirm_period: 24 * HOURS,
+			min_enactment_period: 28 * DAYS,
+			min_approval: APP_MEDIUM_SPENDER,
+			min_support: SUP_MEDIUM_SPENDER,
+		},
+	),
+	(
+		18,
+		pallet_referenda::TrackInfo {
+			name: "big_spender",
+			max_deciding: 10,
+			decision_deposit: 500 * KILOUNIT * SUPPLY_FACTOR,
+			prepare_period: 4,
+			decision_period: 28 * DAYS,
+			confirm_period: 48 * HOURS,
+			min_enactment_period: 28 * DAYS,
+			min_approval: APP_BIG_SPENDER,
+			min_support: SUP_BIG_SPENDER,
+		},
+	),
+];
+
+pub struct TracksInfo;
+impl pallet_referenda::TracksInfo<Balance, BlockNumber> for TracksInfo {
+	type Id = u16;
+	type Origin = <Origin as frame_support::traits::OriginTrait>::PalletsOrigin;
+	fn tracks() -> &'static [(Self::Id, pallet_referenda::TrackInfo<Balance, BlockNumber>)] {
+		&TRACKS_DATA[..]
+	}
+	fn track_for(id: &Self::Origin) -> Result<Self::Id, ()> {
+		if let Ok(system_origin) = frame_system::RawOrigin::try_from(id.clone()) {
+			match system_origin {
+				frame_system::RawOrigin::Root => Ok(0),
+				_ => Err(()),
+			}
+		} else if let Ok(custom_origin) = origins::Origin::try_from(id.clone()) {
+			match custom_origin {
+				origins::Origin::WhitelistedCaller => Ok(1),
+				// General admin
+				origins::Origin::Treasurer => Ok(10),
+				origins::Origin::IdentityAdmin => Ok(11),
+				// Referendum admins
+				origins::Origin::ReferendumCanceller => Ok(12),
+				origins::Origin::ReferendumKiller => Ok(13),
+				// Limited treasury spenders
+				origins::Origin::SmallTipper => Ok(14),
+				origins::Origin::BigTipper => Ok(15),
+				origins::Origin::SmallSpender => Ok(16),
+				origins::Origin::MediumSpender => Ok(17),
+				origins::Origin::BigSpender => Ok(18),
+			}
+		} else {
+			Err(())
+		}
+	}
+}
+
+#[test]
+#[should_panic] // comment out to see curve info for all tracks
+fn print_all_approval_and_support_curves() {
+	// decision_period 28 days
+	for (_, track_info) in TRACKS_DATA {
+		println!("TRACK NAME: {}", track_info.name);
+		println!("Min approval info:");
+		let decision_period_days = track_info.decision_period / DAYS;
+		track_info
+			.min_approval
+			.info(decision_period_days, track_info.name);
+		println!("Min support info:");
+		track_info
+			.min_support
+			.info(decision_period_days, track_info.name);
+	}
+	assert!(false);
```

Comment:
why does this need to panic?

Reply:
It is annotated by `#[should_panic]`. Then I comment that line out to get the curve evolution for all tracks (the curve evolution output does not print unless the test fails).

Reply:
If anyone has suggestions on how to better organize this, let me know. I think it's fine for now.